tests/jax/components/executing/observing_test.py

```diff
@@ -0,0 +1,272 @@
+        executor_parameter_client = MockExecutorParameterClient()
+        # Store
+        store = SimpleNamespace(
+            is_evaluator=None,
+            observations={},
+            policy={},
+            agent_net_keys=agent_net_keys,
+            network_int_keys_extras=network_int_keys_extras,
+            network_sampling_setup=network_sampling_setup,
+            net_keys_to_ids=net_keys_to_ids,
+            timestep=timestep,
+            extras=extras,
+            adder=adder,
+            next_extras=extras,
+            next_timestep=timestep,
+            actions_info=actions_info,
+            policies_info=policies_info,
+            executor_parameter_client=executor_parameter_client,
+        )
+        self.store = store
+
+
+@pytest.fixture
+def mock_executor() -> MockExecutor:
+    """Mock executor component."""
+    return MockExecutor()
+
+
+@pytest.fixture
+def mock_feedforward_executor_observe() -> FeedforwardExecutorObserve:
+    """Mock FeedforwardExecutorObserve.
+
+    Returns:
+        FeedforwardExecutorObserve
+    """
+    return FeedforwardExecutorObserve()
+
+
+# Test on_execution_observe_first
+def test_on_execution_observe_first_without_adder(
+    mock_feedforward_executor_observe: FeedforwardExecutorObserve,
+    mock_executor_without_adder: Executor,
+) -> None:
+    """Test entering executor without store.adder
+
+    Args:
+        mock_feedforward_executor_observe: FeedForwardExecutorObserve,
+        mock_executor_without_adder: Executor
+    """
+    mock_feedforward_executor_observe.on_execution_observe_first(
+        executor=mock_executor_without_adder
+    )
+
+    assert not mock_executor_without_adder.store.adder
+
+
+def test_on_execution_observe_first(
+    mock_feedforward_executor_observe: FeedforwardExecutorObserve,
+    mock_executor: MockExecutor,
+) -> None:
+    """Test on_execution_observe_first method from FeedForwardExecutorObserve
+
+    Args:
+        mock_feedforward_executor_observe: FeedForwardExecutorObserve,
+        mock_executor: Executor
+    """
+    mock_feedforward_executor_observe.on_execution_observe_first(executor=mock_executor)
+
+    assert not mock_executor.store.network_int_keys_extras == None
+    assert (
+        mock_executor.store.extras["network_int_keys"]
+        == mock_executor.store.network_int_keys_extras
+    )
+    assert mock_executor.store.adder.parm == "after_add_first"
+
+
+# test on_execution_observe
+def test_on_execution_observe_without_adder(
+    mock_feedforward_executor_observe: FeedforwardExecutorObserve,
+    mock_executor_without_adder: Executor,
+) -> None:
+    """Test entering executor without store.adder
+
+    Args:
+        mock_feedforward_executor_observe: FeedForwardExecutorObserve,
+        mock_executor_without_adder: Executor
+    """
+    mock_feedforward_executor_observe.on_execution_observe(
+        executor=mock_executor_without_adder
+    )
+
+    assert not mock_executor_without_adder.store.adder
```

Comment:
Likewise to above, this assertion also tests nothing about `on_execution_observe`

Reply:
My suggestion here is similar to above.